pkgs/development/python-modules/watchdog/default.nix

```diff
@@ -32,37 +32,62 @@ buildPythonPackage rec {
     CoreServices
   ];
 
-  propagatedBuildInputs = [
-    pathtools
+  passthru.optional-dependencies.watchmedo = [
     pyyaml
   ];
 
   nativeCheckInputs = [
+    eventlet
     flaky
     pytest-timeout
     pytestCheckHook
-  ];
+  ] ++ passthru.optional-dependencies.watchmedo;
 
   postPatch = ''
     substituteInPlace setup.cfg \
       --replace "--cov=watchdog" "" \
       --replace "--cov-report=term-missing" ""
   '';
 
-  disabledTests = [
-    # probably failing because of an encoding related issue
-    "test_create_wrong_encoding"
-  ] ++ lib.optionals (stdenv.isDarwin && !stdenv.isAarch64) [
-    "test_delete"
-    "test_separate_consecutive_moves"
+  pytestFlagsArray = [
   ] ++ lib.optionals (stdenv.isAarch64) [
-    # fails on aarch64-linux on hydra
-    "test_close"
+    "--deselect=tests/test_emitter.py::test_close"
+  ] ++ lib.optionals (stdenv.isDarwin) [
+    # fails to stop process in teardown
+    "--deselect=tests/test_0_watchmedo.py::test_auto_restart_subprocess_termination"
+    # assert cap.out.splitlines(keepends=False).count('+++++ 0') == 2 != 3
+    "--deselect=tests/test_0_watchmedo.py::test_auto_restart_on_file_change_debounce"
+  ] ++ lib.optionals (stdenv.isDarwin && stdenv.isx86_64) [
+    # FileCreationEvent != FileDeletionEvent
+    "--deselect=tests/test_emitter.py::test_separate_consecutive_moves"
+    # segfaults
+    "--deselect=tests/test_delayed_queue.py::test_delayed_get"
+    "--deselect=tests/test_emitter.py::test_delete"
+  ] ++ lib.optionals (stdenv.isDarwin && stdenv.isAarch64) [
+    # segfaults
+    "--deselect=tests/test_delayed_queue.py::test_delayed_get"
+    "--deselect=tests/test_0_watchmedo.py::test_tricks_from_file"
+    "--deselect=tests/test_fsevents.py::test_watcher_deletion_while_receiving_events_1"
+    "--deselect=tests/test_fsevents.py::test_watcher_deletion_while_receiving_events_2"
+    "--deselect=tests/test_skip_repeats_queue.py::test_eventlet_monkey_patching"
+    "--deselect=tests/test_fsevents.py::test_recursive_check_accepts_relative_paths"
+    # fsevents:fsevents.py:318 Unhandled exception in FSEventsEmitter
+    "--deselect=tests/test_fsevents.py::test_watchdog_recursive"
+    # SystemError: Cannot start fsevents stream. Use a kqueue or polling observer...
+    "--deselect=tests/test_fsevents.py::test_add_watch_twice"
+    # fsevents:fsevents.py:318 Unhandled exception in FSEventsEmitter
+    "--deselect=ests/test_fsevents.py::test_recursive_check_accepts_relative_paths"
```

Comment:
There seems to be a typo in the path. Is this deselection effective?

Reply:
Unlikely